hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -180,11 +185,16 @@ private void addResolvedEntityInsertAction(AbstractEntityInsertAction insert) {
 		}
 		else {
 			LOG.trace( "Adding resolved non-early insert action." );
-			insertions.add( insert );
+			if(insertions == null) {
+				insertions = new ExecutableList<AbstractEntityInsertAction>( new InsertActionSorter() );
+			}
+			insertions.add(insert);
 		}
 		insert.makeEntityManaged();
-		for ( AbstractEntityInsertAction resolvedAction : unresolvedInsertions.resolveDependentActions( insert.getInstance(), session ) ) {
-			addResolvedEntityInsertAction( resolvedAction );
+		if(unresolvedInsertions != null) {
+			for (AbstractEntityInsertAction resolvedAction : unresolvedInsertions.resolveDependentActions(insert.getInstance(), session)) {
+				addResolvedEntityInsertAction(resolvedAction);
+			}
```

Comment:
Hi Stuart,
I think your patch violates the code style used by Hibernate at multiple places. Here is for example how it should look like:
```
	if ( unresolvedInsertions != null ) {
		for ( AbstractEntityInsertAction resolvedAction : unresolvedInsertions.resolveDependentActions( insert.getInstance(), session ) ) {
			addResolvedEntityInsertAction( resolvedAction );
		}
	}
```